uploaded watering can updates for fluffymachines.

Rene98 opened the pull request.

Rene98 committed:
uploaded watering can updates for fluffymachines.

NCBPFluffyBear commented:
This should be coded to work with this plugin, not Fluffymachines

Rene98 committed:
watering can in supserv

NCBPFluffyBear commented on src/main/java/io/ncbpfluffybear/supserv/items/WateringCan.java:
```diff
@@ -0,0 +1,289 @@
+  private static final int MAX_SUGAR_GROW_HEIGHT = 5;
+  private static final NamespacedKey usageKey = new NamespacedKey(SupServPlugin.getInstance(), "watering_can_usage");
+
+  private final canType canType;
```
Move private variables under class constants

NCBPFluffyBear replied:
The can type is also never needed, just the size. You can instead create a private variable for the integer size and then assign it in the constructor.

NCBPFluffyBear commented on src/main/java/io/ncbpfluffybear/supserv/items/WateringCan.java:
```diff
@@ -0,0 +1,289 @@
+                     ItemStack[] recipe, canType size) {
+    super(category, item, recipeType, recipe);
+    this.canType = size;
+    addItemSetting(maxUses);
```
You have this ItemSetting, but it seems that it doesn't actually affect the size

Rene98 replied:
Your totally right, this is not needed anymore since enum of watercanTypes

Rene98 committed:
relocated class constants

NCBPFluffyBear commented:
One more thing: Can you add some kind of prefix or suffix to the IDs? Right now, there is an item registration conflict because the same Watering can ID is used for SupServ and FM

NCBPFluffyBear commented:
There are still some conflicts, please fix

NCBPFluffyBear committed:
Fix conflicts

NCBPFluffyBear committed:
Merge remote-tracking branch 'origin/watering-can' into watering-can

NCBPFluffyBear committed:
Merge branch 'master' into watering-can

NCBPFluffyBear commented:
> One more thing: Can you add some kind of prefix or suffix to the IDs? Right now, there is an item registration conflict because the same Watering can ID is used for SupServ and FM

This is still an issue, once the ID conflict occurs, FM is disabled. Discussed this issue with Skizz and he decided it would be best if there was an additional recipe using the FM watering can to craft the tier 2 SS can

Rene98 committed:
Solved conflict with fluffy machines watering can
added recipe for FM watering can to be crafted into SupServ watering can [iron]

Rene98 committed:
Merge remote-tracking branch 'origin/watering-can' into watering-can

# Conflicts:
#	src/main/java/io/ncbpfluffybear/supserv/SupServItems.java

NCBPFluffyBear commented:
Do you think we should have a label on the second recipe saying it uses the FM watering can to avoid confusion?
Or, we could have skizzles just hide the recipe from the guide

NCBPFluffyBear commented:
Ok skizz said that he would hide it. In that case, could you change the lore so it doesnt look like a watering can? like put some kind of tag saying it shouldnt be obtainable

Rene98 committed:
Changed desc for watering can conversion recipes to is only for conversion use

NCBPFluffyBear commented:
Resolve conflicts

Skizzles commented:
There is also an error with this area:
https://screenshots.skizzles.me/wAnA3/peBIsOrU75.png

This error stops the upgrading still as this recipe does not register.

Rene98 committed:
added FM to softdepend

Rene98 committed:
Merge branch 'master' of https://github.com/Rene98/SupplementalServiceableness into watering-can

 Conflicts:
	src/main/java/io/ncbpfluffybear/supserv/SupServItemSetup.java
	src/main/java/io/ncbpfluffybear/supserv/SupServItems.java

Rene98 committed:
Fixed conflicts

NCBPFluffyBear committed:
Merge branch 'watering-can' of https://github.com/Rene98/SupplementalServiceableness into watering-can

NCBPFluffyBear committed:
Fixed compressed cobblestone issues and changed some formatting
